sphinx/builders/linkcheck.py

```diff
@@ -65,6 +66,17 @@
 DEFAULT_DELAY = 60.0
 
 
+class LinkStatus(StrEnum):
+    BROKEN = 'broken'
+    IGNORED = 'ignored'
+    TIMEOUT = 'timeout'
+    RATE_LIMITED = 'rate-limited'
+    REDIRECTED = 'redirected'
+    UNCHECKED = 'unchecked'
+    UNKNOWN = 'unknown'
```

Comment:
@AA-Turner I'd considered making `unknown` a first-class literal of its own, as opposed to creating a type alias for empty-string.  I don't have a strong preference either way, though.

(in some ways, in fact, I might prefer a third way - using `None` instead and making the value optional in that one limited code site)

Reply:
The following may be of interest:

* https://mypy.readthedocs.io/en/stable/literal_types.html
* https://typing.readthedocs.io/en/latest/spec/literal.html

Notably, there doesn't seem much point to defining a `Literal` of every member of an enum. But in the `timeout_status` case, it could be useful.

A

Reply:
Thank you!  I'll try to apply some of that guidance soon (including the [special-casing of `None`](https://typing.readthedocs.io/en/latest/spec/literal.html#legal-parameters-for-literal-at-type-check-time), if possible..).